### ISSUE TITLE
[FE] Convert profile phone number form to use SchemaForm

### ISSUE THREAD

**Issue by Samara-Strauss** [department-of-veterans-affairs/va.gov-team#5118]
## Background

Epic #4068

## Tasks
- [ ] Make a new version of the phone number component that wraps a `SchemaForm` component and does not use any `Errorable` form components

## Acceptance Criteria:
- [ ] Should function the same as the existing phone number component.
- [ ] Needs to be able to live alongside the existing phone number form (ie, behind a feature flag or some other method so the form can be put on staging but not prod)

**Comment by Samara-Strauss** [comment]
@erikphansen this is still in progress, yes? I have it in the Sprint 15 objectives and want to make sure that's accurate.

**Comment by erikphansen** [comment]
There is a PR open for this that's waiting on VSP review which should be done today. https://github.com/department-of-veterans-affairs/vets-website/pull/11677

**Comment by Samara-Strauss** [comment]
OK, thanks for letting me know!

**Comment by erikphansen** [comment]
Merged. I moved this to Validate and then we can close when we confirm editing phone numbers still works.

**Comment by Samara-Strauss** [comment]
So I noticed a couple issues with the phone number form:

1. The form is requiring an extension even though an extension should not be required (also, what is with that weird error message?).
2. Also, there is a weird spacing issue with the information "i" icon and the copy in the alert box, but that may be a separate issue related to #5553 
<img width="442" alt="Screen Shot 2020-02-12 at 3 52 55 PM" src="https://user-images.githubusercontent.com/34068740/74376464-28a97b00-4db0-11ea-8944-b62e773ee628.png">
3. When we originally built the form, we represented the "1" for the country code as follows. Idk where it went but it should be in the form. Here's an old mock to show what that looked like (please ignore the broken icon that's showing as a "?"):
<img width="336" alt="Screen Shot 2020-02-12 at 3 55 08 PM" src="https://user-images.githubusercontent.com/34068740/74376605-745c2480-4db0-11ea-9bc9-d9fc59c44a66.png">

Idk if any of these issues are related to your work but this seemed as good a place as any to raise the flag about them.

**Comment by erikphansen** [comment]
Thanks for catching this, @Samara-Strauss. This was super sloppy on my part 😞 

I have a PR to fix items 1 and 2 (plus another issue I noticed when I went in to fix those): https://github.com/department-of-veterans-affairs/vets-website/pull/11685


I'm not sure about item 3. This is new to me. And it's not like that on prod right now. @ncksllvn do you know anything about "hardcoding" the +1 in the input field like that? Was that ever implemented? Regardless, with the move to a SchemaForm for the phone number form, that _currently_ won't be possible, as far as I know. But would be a fun thing to add to the forms system. Let me know how badly you want me to pursue that, Samara!

**Comment by Samara-Strauss** [comment]
> What version of Chrome are you using? This is... weird.

I'm on version 80.0.3987.100. Idk if that's the most up to date one or not, but I can tell you I'm also on iOS High Sierra because I have to be because the VA VPN won't work on anything newer (I know, I know 🙄). Idk if that would have any effect on Chrome or how things render — I would think not — but maybe.

> I'm not sure about item 3. This is new to me. And it's not like that on prod right now. @ncksllvn do you know anything about "hardcoding" the +1 in the input field like that? Was that ever implemented? Regardless, with the move to a SchemaForm for the phone number form, that currently won't be possible, as far as I know. But would be a fun thing to add to the forms system. Let me know how badly you want me to pursue that, Samara!

If it's not possible with SchemaForm then I wouldn't immediately worry about it. Not worth jumping through hoops of fire to implement this.

**Comment by erikphansen** [comment]
@Samara-Strauss this should be working correctly now.

**Comment by ncksllvn** [comment]
> do you know anything about "hardcoding" the +1 in the input field like that?

I do remember. At the time, it was implemented in a very temporary way, because our Form System's input components didn't support the addition of the gray box indicating the hardcoded country code. It was removed later because of a strange incompatibility identified while we were upgrading our React version, that was causing a browser error. I think we wrote a ticket in vets.gov-team for adding the number back. Sorry about this.

**Comment by erikphansen** [comment]
I should probably make a custom form system widget to implement this. Since I don't know how to do that but it would probably be pretty easy as far as custom widgets go. @Samara-Strauss feel free to make a ticket to create a new custom `DomesticPhoneNumber` widget if you think is worth spending a couple points on.

**Comment by Samara-Strauss** [comment]
> @Samara-Strauss this should be working correctly now.

For the Home, Work, and Fax numbers, the "Update" button doesn't work both when I don't change the phone number and when I do.

The Update buttons is working for the Mobile phone field, but I am wondering if that's because I checked the option for notifications.

For the Fax number, I was able to add a number, but I wasn't able to update it after I saved it the first time.

**Comment by erikphansen** [comment]
Yeah there's all sorts of weird stuff going on with the new telephone forms. Looking into that now.

**Comment by erikphansen** [comment]
@Samara-Strauss this should now work on staging. However I did uncover another weird bug tracked here: https://github.com/department-of-veterans-affairs/va.gov-team/issues/5963

**Comment by Samara-Strauss** [comment]
Awesome. This bug is all set. Can we close this or do you want to wait until #5963 is done?

**Comment by erikphansen** [comment]
Let's close this
